src/services/createPropStore.ts

```diff
@@ -111,6 +114,59 @@ export function createPropStore(): PropStore {
     return store.get(key)
   }
 
+  function getParentContext(route: ResolvedRoute, prefetch: boolean = false): PropsCallbackParent {
+    const parent = route.matches.at(-2)
+
+    if (!parent) {
+      return
+    }
+
+    if (isWithComponentProps(parent)) {
+      return {
+        name: parent.name ?? '',
+        get props() {
+          return getParentProps(parent, 'default', route, prefetch)
+        },
```

Comment:
I made this a get so that the prefetch warning wouldn't happen unless you actually tried to access the props. But I need to figure out how to test that.